### PR TITLE
fix($compile): Fix namespace detection for achor elements

### PR DESCRIPTION
**Pull request by feelepxyz**
Fixed a bug where anchor elements with `SVG` in the href attribute got incorrectly detected as svgs. Seems the detection only happens when the anchor element has a child that needs to be transcluded.

Calling `node.toString()` on a anchor returns the `href` value instead of the expected `[object HTMLAnchorElement]`.

Added a test for this edge case.

Plunker to reproduce: http://plnkr.co/edit/OBgCS8phjTwMTFYwhk6D

### CODE REVIEW

File: src/ng/compile.js [hunk 1]
```diff
@@ -1396,7 +1396,7 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
       if (!node) {
         return 'html';
       } else {
-        return nodeName_(node) !== 'foreignobject' && node.toString().match(/SVG/) ? 'svg' : 'html';
+        return nodeName_(node) !== 'foreignobject' && Object.prototype.toString.call(node).match(/SVG/) ? 'svg' : 'html';
```
Comment (lgalfaso) [inline src/ng/compile.js]: Just
```js
return nodeName_(node) !== 'foreignobject' && toString.call(node).match(/SVG/) ? 'svg' : 'html';
```

**Comment by harrison** [comment]
@lgalfaso sorted. Added a commit for the same fix in ngInclude, but looking at that code, the function `namespaceAdaptedClone` doesn't seem to be defined anywhere in the codebase.. is this directive ever used? :)

**Comment by lgalfaso** [comment]
@harrison the only reference of `namespaceAdaptedClone` is within `src/ng/ngInclude.js`, do not know how this is related to this PR.

One more thing, can you please name the test something like "should handle anchor elements inside svg elements"?

**Comment by harrison** [comment]
@lgalfaso not related just curious if it's useless code:)

The issue happens with the ngIf/transclusion on the `span` which wrongly detects the parent anchor as being an svg. Changed it to `transclusions should handle parent anchor elements`, happy to change.

**Comment by lgalfaso** [comment]
@harrison can you please squash everything into one commit? this makes things easier. Otherwise this looks good and should merge early next week
